core/lib/sync/task.ts

```diff
@@ -53,11 +53,11 @@ function getPayload (task, currentToken) {
   const payload = {
     mode: 'cors',
     method: 'GET',
-    ...task.payload,
+    ...(task && task.payload ? task.payload : {}),
```

Comment:
`...(task?.payload || {})` maybe use optiona chaining, wdyt? Same for second change